packages/react-jss/src/createHoc.js

```diff
@@ -176,25 +199,15 @@ export default (stylesOrCreator, InnerComponent, options = {}) => {
         })
       }
 
-      const defaultClasses = InnerComponent.defaultProps
-        ? InnerComponent.defaultProps.classes
-        : undefined
-      const jssClasses = dynamicSheet
-        ? compose(
-            staticSheet.classes,
-            dynamicSheet.classes
-          )
-        : staticSheet.classes
-      const classes = {
-        ...defaultClasses,
-        ...jssClasses,
-        ...userClasses
-      }
+      const sheet = dynamicSheet || staticSheet
+      // $FlowFixMe defaultProps are not defined in React$Component
+      const defaultClasses = InnerComponent.defaultProps ? InnerComponent.defaultProps.classes : {}
+      const classes = {...defaultClasses, ...sheet.classes, ...userClasses}
 
```

Comment:
Why did you change the logic of this code?

Reply:
For now this is essentially a copy and paste from previous PR. Given the diff shows there are some changes (such as this one) that haven't been integrated in it, I'll take care of it top-priority, before working on anything else

Reply:
Reverted